src/utils/utils.ts

```diff
@@ -31,3 +32,15 @@ export async function executeCommand(command: string, cwd?: string): Promise<str
 export function generateQuotedFullName(schema: string, objectName: string): string {
 	return `[${escapeClosingBrackets(schema)}].[${escapeClosingBrackets(objectName)}]`;
 }
+
+/**
+ * Promise with timeout to check for another promise fulfillment
+ * @returns a promise that resolves after the specified timeout
+ */
+export function waitForPromise(): Promise<void> {
```

Comment:
I'd suggest just `timeoutPromise` or something similar - this function isn't actually doing any "waiting" so naming it `waitFor...` isn't really accurate